--- a/configure.py
+++ b/configure.py
@@ -19,13 +19,13 @@
 # header
 
 print( '-'*80 + '\n' +
-font.bold( font.blue( '                              Welcome to BLAS++.' ) ) +
+font.bold( font.cyan( '                              Welcome to BLAS++.' ) ) +
 '''
 
 By default, configure will automatically choose the first valid value it finds
 for each option. You can set it to interactive to find all possible values and
 give you a choice:
-    ''' + font.blue( 'make config interactive=1' ) + '''
+    ''' + font.cyan( 'make config interactive=1' ) + '''
 
 If you have multiple compilers, we suggest specifying your desired compiler by
 setting CXX, as the automated search may prefer a different compiler.
--- a/config/config.py
+++ b/config/config.py
@@ -205,8 +205,8 @@
     '''
     if (label):
         if (rc == 0):
-            print( font.blue( 'yes'  ), extra, file=log )
-            print( font.blue( ' yes' ), extra )
+            print( font.cyan( 'yes'  ), extra, file=log )
+            print( font.cyan( ' yes' ), extra )
         else:
             print( font.red( 'no'  ), extra, file=log )
             print( font.red( ' no' ), extra )
